src/Raven.Client/ServerWide/Tcp/TcpConnectionHeader.cs

```diff
@@ -25,14 +26,44 @@ public enum OperationTypes
         public int OperationVersion { get; set; }
 
         public string Info { get; set; }
-
-        public static readonly int NumberOfRetriesForSendingTcpHeader = 2;
+      
         public static readonly int ClusterTcpVersion = 10;
         public static readonly int HeartbeatsTcpVersion = 20;
-        public static readonly int ReplicationTcpVersion = 31;
+        public static readonly int ReplicationTcpVersion = 33;
         public static readonly int SubscriptionTcpVersion = 40;
         public static readonly int TestConnectionTcpVersion = 50;
 
+        public static (bool Supported, int PrevSupported) OperationVersionSupported(OperationTypes operationType, int version)
+        {
+            var prev = -1;
+            if (_operationsToSupportedProtocolVersions.TryGetValue(operationType, out var supportedProtocols) == false)
+                return (false, prev);
+            
+            for (var i =0; i< supportedProtocols.Count; prev = supportedProtocols[i], i++)
+            {
+                var current = supportedProtocols[i];
+                if (current == version)
+                    return (true, prev);
+
+                if (current > version)
+                    return (false, prev);
+            }
+
+            return (false, prev);
+        }
+
+        private static readonly Dictionary<OperationTypes, List<int>> _operationsToSupportedProtocolVersions
+        = new Dictionary<OperationTypes, List<int>>
+            {
+                [OperationTypes.Ping] = new List<int> { -1},
+                [OperationTypes.None] = new List<int> { -1 },
+                [OperationTypes.Drop] = new List<int> { -2 },
+                [OperationTypes.Subscription] = new List<int> { 40 },
+                [OperationTypes.Replication] = new List<int> { 33, 31 },
```

Comment:
I suggest bumping those to:
- 40xxx for 4.0 supported protocols
- 41xxx for 4.1 supported protocols

This way it would be much more clear which protocol is related to which 4.x version.

Reply:
Not binding it to a server version. Just making clear which protocol is supported where.

Image a list of protocols like that:
`32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43`

Which ones are supported by 4.0, which ones are supported by 4.1 and which ones are supported by 4.2?

If this list would look like:

`40032, 40033, 41001, 42010, 41002, 40035, 42015, 41032`

Then it would be more clear which protocol is from which version.